### ISSUE TITLE
React | Tabbar issue: Missing Tab bar icon when we display a badge using state (real time)

### ISSUE THREAD

**Issue by rj-david** [OnsenUI/OnsenUI#2693]
__Environment__

  
```
[Core]
  onsenui latest
  
[Framework]
  react 15.4.2

[Framework binding]
 react-onsenui latest

[Platform]
  Desktop - Linux 64-bit

[Browser]
  Desktop - Chrome 76.0.3809.132 on Linux 64-bit
```

__Encountered problem__
  When we display a badge using state (from no badge to with badge), the tab bar icon is being hidden (missing). This is a problem with onsenui 2.10.10

NOT an issue with onsensui 2.10.8

__How to reproduce__
In this example, click the Add Badge text in the right of the header. The icon is missing in the tab bar.

  
  [__Demo link__](https://onsen.io/playground/?issue)

  

- __HTML__

```html
<div id="app"></div>
```

- __JS__

```jsx
var MyTab = React.createClass({
  render: function() {
    return (
      <Ons.Page>
        <section style={{margin: '16px'}}>
          <p>
            {this.props.content}.
          </p>
        </section>
      </Ons.Page>
    );
  }
});

var MyPage = React.createClass({
  getInitialState: function() {
    return {
      index: 0,
      badge: 0
    }
  },
  addBadge: function() {
    this.setState(this.setState({badge: 1}));
  },
  renderToolbar: function() {
    const titles = ['Home', 'Settings'];
    return (
      <Ons.Toolbar>
        <div className='center'>{titles[this.state.index]}</div>
        <div className='right' onClick={this.addBadge.bind(this)}>Add Badge</div>
      </Ons.Toolbar>
    );
  },
  renderTabs: function() {
    return [
      {
        content: <MyTab content="Welcome home" />,
        tab: <Ons.Tab label='Home' icon='md-home' badge={this.state.badge || null} />
      },
      {
        content: <MyTab content="Change the settings" />,
        tab: <Ons.Tab label='Settings' icon='md-settings' />
      }
    ];
  },

  render: function() {
    return (
      <Ons.Page renderToolbar={this.renderToolbar}>
        <Ons.Tabbar
          swipeable={true}
          position='auto'
          index={this.state.index}
          onPreChange={(event) =>
            {
              if (event.index != this.state.index) {
                this.setState({index: event.index});
              }
            }
          }
          renderTabs={this.renderTabs}
        />
      </Ons.Page>
    );
  }
});

ons.ready(function() {
  ReactDOM.render(<MyPage />, document.getElementById('app'));
});
```

**Comment by rj-david** [comment]
@emccorson, sorry for the incomplete information on my other issue ticket. Did not realize that there was a change in icon names. There is an issue with the tab bar icon when we change the state of badge real time i.e. when the tab bar is already mounted. The bug exists in 2.10.10 but not in 2.10.8 (might be a regression with the tab-bar related fixes in 2.10.9)

**Comment by TF-Jonny** [comment]
Had the same issue. After adding the same icon also to the active icon attribute on the tabbar this seemed to resolve the vanishing icon whilst populating a badge on the fly.

HTML
<ons-tab id="tabbar1" label="tabbar" icon="md-assignment-o" active-icon="md-assignment-o"" active>
</ons-tab>

Javascript
document.querySelector("#tabbar1").setAttribute("badge", 99);

**Comment by rkgrep** [comment]
Same issue with vue version.
Even official docs page has the bug present
![Screenshot-2020-01-04-001](https://user-images.githubusercontent.com/2423048/71768589-ef2c5700-2f49-11ea-8c6d-9f10d3ce3644.png)

**Comment by terphic** [comment]
Has anyone here figured out a workaround in the meantime? I'm currently having this issue as well. Adding ActiveIcon did not work for me (vue.js).

**Comment by emccorson** [comment]
This seems to be fixed already. I think probably it was fixed by #2720. In any case, the example provided doesn't show the problem anymore.

@rkgrep This seems to be a different problem. I think in this case, the examples in the documentation just haven't been updated yet since the names of the icons changed in 2.10.9.
